tests/unit_tests/cpu/CMakeLists.txt

```diff
@@ -15,38 +15,89 @@ detray_add_test( cpu
 macro( detray_add_cpu_test algebra )
    # Build the test executable.
    detray_add_test( cpu_${algebra}
-      "axis_rotation.cpp"
-      "tools_bounding_volume.cpp"
+      "../axis_rotation.cpp"
```

Comment:
hmm why has the directory changed? I prefer the before

Reply:
So, I found the reason for the issue that kept making the tests fail. If all of the simulation tests from the different plugins are built in the same directory then some files will get reused improperly.
This is probably not the correct way to do it, but this is just an attempt to fix it by creating the different tests in different directories. I tried to figure out how to change this CMake macro create a directory and build there but without success so far

Reply:
Hm, is is not possible to just ask the simulator to make an output directory and write the files into that?

Reply:
You can set it in the simulator
```c++
    // `check_simulation.inl`
    auto sim = simulator(n_events, detector, std::move(generator), smearer, "put directory here");
```
BTW can we set the output directory differently for every plugins